acados/utils/types.h

```diff
@@ -35,6 +35,21 @@
 #ifndef ACADOS_UTILS_TYPES_H_
 #define ACADOS_UTILS_TYPES_H_
 
+/* Symbol visibility in DLLs */
```

Comment:
I see that you adapted this part from the `CasADi` exported functions.

`GCC_HASCLASSVISIBILITY` seems to be SWIG related https://github.com/swig/swig/blob/b4a92ae34f6d39825bc0eba056d6983575db4695/Lib/swiglabels.swg#L71
So I am not sure, if this is relevant / would work here.
Maybe these conditions should be checked instead.
https://github.com/swig/swig/blob/b4a92ae34f6d39825bc0eba056d6983575db4695/Lib/swiglabels.swg#L69

If you have a better understanding of this, a comment would be nice.

Reply:
It can be that I "borrowed" this from CasADi (it is already some time ago). I probably thought if it works there it should work here as well. There is not more behind it. The main part for MSVC compatibility is
`#define ACADOS_SYMBOL_EXPORT __declspec(dllexport)`
I can remove lines 46 and 47, if you want.